frontend/package.json

```diff
@@ -144,51 +133,65 @@
     ]
   },
   "devDependencies": {
+    "@babel/preset-env": "^7.24.7",
+    "@babel/preset-typescript": "^7.24.7",
     "@fortawesome/fontawesome-common-types": "^6.5.1",
     "@react-pdf/types": "^2.1.0",
-    "@testing-library/dom": "^6.15.0",
-    "@testing-library/jest-dom": "^4.2.4",
-    "@testing-library/react": "^9.3.2",
-    "@testing-library/user-event": "^7.1.2",
-    "@turf/helpers": "^6.5.0",
+    "@testing-library/dom": "^10.3.1",
+    "@testing-library/jest-dom": "^6.1.3",
+    "@testing-library/react": "^16",
+    "@testing-library/user-event": "^14.5.2",
+    "@turf/helpers": "^7",
     "@types/chart.js": "^2.9.21",
     "@types/colormap": "^2.3.1",
     "@types/d3": "^5.7.2",
     "@types/dompurify": "^2.2.3",
     "@types/geojson": "^7946.0.7",
-    "@types/jest": "^24.0.0",
+    "@types/jest": "^29.5.4",
     "@types/lodash": "^4.14.149",
     "@types/marked": "^4.0.2",
-    "@types/node": "^16.0.0",
+    "@types/node": "20.8.8",
     "@types/papaparse": "^5.0.3",
-    "@types/react": "^16.13.0",
+    "@types/react": "^18",
     "@types/react-datepicker": "^2.11.0",
-    "@types/react-dom": "^16.9.0",
+    "@types/react-dom": "^18",
     "@types/react-pdf": "^5.7.2",
     "@types/react-redux": "^7.1.7",
     "@types/react-router-dom": "^5.1.3",
     "@types/redux-mock-store": "^1.0.6",
+    "@typescript-eslint/eslint-plugin": "^6.0.0",
+    "@typescript-eslint/parser": "^6.0.0",
```

Comment:
I'm seeing this warning when running, did you consider upgrading to v7?
<img width="928" alt="Screenshot 2024-07-09 at 10 20 10 AM" src="https://github.com/WFP-VAM/prism-app/assets/8203830/848f8109-84f5-4cdd-bfcb-1b969cab256d">